chore(website): Update Docs to show information for new Destination Spec option (#9257)
cloudquery/cloudquery: website/pages/docs/reference/destination-spec.md

The max number of bytes to use for a single batch. Only applies to plugins that utilize batching. This setting works in conjunction with `batch_size`, and batches are written whenever either `batch_size` or `batch_size_bytes` is reached. Every plugin has its own default value for `batch_size_bytes`. Note that the size in bytes is calculated based on the size of data in memory, not the serialized data, and it is best to choose a `batch_size_bytes` significantly lower than any hard limits.

<!-- vale off -->
### pk_mode
<!-- vale on -->

> (`string`, optional, default: `default`, Available: `default`, `cq-id-only` introduced in CLI `v2.5.2`)

Author: agreed, while I think that using <date> as a reference point is probably the best thing we can do right now, I think we need a better way of documenting compatibility between CLI, config and plugins

> (`string`, optional, default: `default`, Available: `default`, `cq-id-only` introduced in CLI `v2.5.2`)

Reviewer: Ah, we have the same issue with https://www.cloudquery.io/docs/reference/source-spec#deterministic_cq_id

> (`string`, optional, default: `default`, Available: `default`, `cq-id-only` introduced in CLI `v2.5.2`)

Author: Updated. Added a similar note to the source spec as well

Specifies the Primary Keys that the destination will configure when using the `overwrite` or `overwrite-delete-stale` mode. 

- `default`: The default primary keys are used.

> - `cq-id-only`: The `_cq_id` field is used as the only primary key for each table.  This is useful when you don't want breaking changes to primary keys to impact your schema. It is highly reccomended that if you are using this feature you should also use the [`deterministic_cq_id` feature in the source](docs/reference/source-spec#deterministic_cq_id). If you are using `overwrite` mode and a source updates a primary key, this will result in a new row being inserted. If you are using `overwrite-delete-stale` mode, a new row will be inserted and the old row will be deleted as a stale resource. Note: using this parameter might result in changes to query performance as queries will not have the ability to use indexes.  

Reviewer: ```suggestion
- `cq-id-only`: The `_cq_id` field is used as the only primary key for each table.  This is useful when you don't want breaking changes to primary keys to impact your schema. It is highly recommended that if you are using this feature you should also use the [`deterministic_cq_id` feature in the source](docs/reference/source-spec#deterministic_cq_id). If you are using `overwrite` mode and a source updates a primary key, this will result in a new row being inserted. If you are using `overwrite-delete-stale` mode, a new row will be inserted and the old row will be deleted as a stale resource. Note: using this parameter might result in changes to query performance as queries will not have the ability to use indexes.  
```

> - `cq-id-only`: The `_cq_id` field is used as the only primary key for each table.  This is useful when you don't want breaking changes to primary keys to impact your schema. It is highly recommended that if you are using this feature you should also use the [`deterministic_cq_id` feature in the source](docs/reference/source-spec#deterministic_cq_id). If you are using `overwrite` mode and a source updates a primary key, this will result in a new row being inserted. If you are using `overwrite-delete-stale` mode, a new row will be inserted and the old row will be deleted as a stale resource. Note: using this parameter might result in changes to query performance as queries will not have the ability to use indexes.  

Reviewer 2: > Note: using this parameter might result in changes to query performance as queries will not have the ability to use indexes.

Not factually correct. There won't be any indexes due to the fact that the destination won't be creating any.

Author: Updated

<!-- vale off -->
### pk_mode
<!-- vale on -->

(`string`, optional, default: `default`, Available: `default`, `cq-id-only` introduced in CLI `v2.5.2`)

Specifies the Primary Keys that the destination will configure when using the `overwrite` or `overwrite-delete-stale` mode. 

- `default`: The default primary keys are used.
- `cq-id-only`: The `_cq_id` field is used as the only primary key for each table.  This is useful when you don't want breaking changes to primary keys to impact your schema. It is highly recommended that if you are using this feature you should also use the [`deterministic_cq_id` feature in the source](/docs/reference/source-spec#deterministic_cq_id). If you are using `overwrite` mode and a source updates a primary key, this will result in a new row being inserted. If you are using `overwrite-delete-stale` mode, a new row will be inserted and the old row will be deleted as a stale resource. Note: using this parameter might result in changes to query performance as CloudQuery will not be creating indexes for the default primary key columns.  

> The latest version of all destination plugins support this feature, but only destination plugins released after 2023-03-21 will support this functionality. 

Reviewer: ```suggestion
Supported by destination plugins released on 2023-03-21 and later 
```
